examples/SmoothedParticleHydrodynamics/src/model/functions.c

```diff
@@ -0,0 +1,233 @@
+ *
+ * University of Sheffield retain all intellectual property and
+ * proprietary rights in and to this software and related documentation.
+ * Any use, reproduction, disclosure, or distribution of this software
+ * and related documentation without an express license agreement from
+ * University of Sheffield is strictly prohibited.
+ *
+ * For terms of licence agreement please attached licence or view licence
+ * on www.flamegpu.com website.
+ *
+ */
+
+/* 
+	Smoothed Particle Hydrodynamics Demo
+	Author: Matthew Leach
+	Contact: M.Leach@sheffield.ac.uk
+
+	Suggested things to try:
+	
+	Play with the viscosity to try fluids of different thickness - the simulation is
+	stable for viscosity values between ~0.1 - 30
+
+	Lower the fluid rest density to TODO see the 'FLAME' explode instead of fall
+
+	Use the python script in the iterations folder to change the initial arrangement of the fluid particles
+*/
+
+
+#ifndef _FLAMEGPU_FUNCTIONS
+#define _FLAMEGPU_FUNCTIONS
+
+#include <header.h>
+
+// Mathematic constants
+#define PI 3.141593f
+
+// Sim parameters
+#define TIMESTEP 0.001f
+#define SMOOTHING_LENGTH 0.040f
+#define MAX_VELOCITY 4.0f
+
+// Environemental parameters
+#define RESTITUTION_COEFFICIENT 0.9f
+#define HALF_BOUNDARY_WIDTH 0.5f
+
+// Fluid properties
+#define PARTICLE_MASS 0.001953125f
+#define PRESSURE_COEFFICIENT 0.9f
+#define FLUID_REST_DENSITY 1000.0f
+#define MINIMUM_PARTICLE_DENSITY 0.0f
+#define VISCOSITY 10.0f
+
+//// Math helpers
+inline __device__ float dot(glm::vec3 a, glm::vec3 b) {
+	return a.x * b.x + a.y * b.y + a.z * b.z;
+}
+
+inline __device__ float length(glm::vec3 v) {
+	return sqrt(dot(v, v));
+}
+
+//// Smoothing functions
+__device__ float computeW(float distance, float smoothingLength) {
+	return distance < smoothingLength ? (315.0f / (64.0f*PI*pow(smoothingLength, 9))) * pow((pow(smoothingLength, 2) - pow(distance, 2)), 3) : 0.0f;
+}
+
+__device__ glm::vec3 computeDelW(float distance, glm::vec3 difference, float smoothingLength) {
+	return distance < smoothingLength ? -(45.0f / (PI*pow(smoothingLength, 6)))*(pow(smoothingLength - distance, 2))* (1.0f / distance) * difference : glm::vec3(0.0f);
+}
+
+__device__ float computeDelSqW(float distance, float smoothingLength) {
+	return distance < smoothingLength ? (45.0f / (PI*pow(smoothingLength, 6))) * (smoothingLength - distance) : 0.0f;
+}
+
+__device__ float computeSurfaceTension(float distance, float smoothingLength) {
+	if (distance > smoothingLength / 2.0f && distance <= smoothingLength)
+		return 10 * (32.0f / (PI*pow(smoothingLength, 6))) * pow(smoothingLength - distance, 3) * pow(distance, 3);
+	if (distance > 0 && distance <= smoothingLength / 2.0f)
+		return (((64.0f / (PI*pow(smoothingLength, 6))) * 2 * pow(smoothingLength - distance, 3) * pow(distance, 3)) - pow(smoothingLength, 4) / 64.0f);
+	return 0.0f;
+}
+
+/**
+ * outputLocationVelocity FLAMEGPU Agent Function
+ * Automatically generated using functions.xslt
```

Comment:
Probably worth either changing the docblock comment to reflect what the function does, or just remove the docblock?